### PR TITLE
Change sqlite index to correct column

### PR DESCRIPTION
**Pull request by jdkuki**
Fixes brave/brave-browser#4663

Database query was using incorrect column index to get `provider` for `publisher_info` object.

image

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
See steps in brave/brave-browser#4663

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by jdkuki** [comment]
@NejcZdovc updated

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 1]
```diff
@@ -744,10 +744,11 @@ bool PublisherInfoDatabase::GetExcludedList(
   }
 
   // We will use every attribute from publisher_info
-  std::string query = "SELECT * FROM publisher_info WHERE excluded = 1";
+  std::string query = "SELECT publisher_id, verified, excluded, "
+                      "name, favicon, url, provider "
+                      "FROM publisher_info WHERE excluded = 1";
 
   sql::Statement info_sql(db_.GetUniqueStatement(query.c_str()));
-
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/publisher_info_database.cc]: nit: please leave this line

File: components/brave_rewards/browser/publisher_info_database.cc [hunk 2]
```diff
@@ -744,10 +744,11 @@ bool PublisherInfoDatabase::GetExcludedList(
   }
 
   // We will use every attribute from publisher_info
-  std::string query = "SELECT * FROM publisher_info WHERE excluded = 1";
+  std::string query = "SELECT publisher_id, verified, excluded, "
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/publisher_info_database.cc]: excluded is not used, so we don't need to select it

Comment (jdkuki) [reply components/brave_rewards/browser/publisher_info_database.cc]: updated

**Comment by jdkuki** [comment]
> Nice find. Please replace `*` with actual columns. This way we can't regress if we change DB structure

Fixed

**Pull request merged**